### PR TITLE
refactor!: Remove `CovarianceTransport`

### PR DESCRIPTION
**Pull request by andiwand**
This seems not to be used anywhere other than benchmarks

Breaking as this is part of the public interface

### CODE REVIEW

**Comment by paulgessinger** [comment]
I think this is used in the EDM4hep plugin at least.

File: Plugins/EDM4hep/src/EDM4hepUtil.cpp [hunk 1]
```diff
@@ -144,25 +145,9 @@ Parameters convertTrackParametersToEdm4hep(const Acts::GeometryContext& gctx,
...
-    Acts::CovarianceCache covCache{freePars.value(), freeCov};
-    auto [varNewCov, varNewJac] = Acts::transportCovarianceToBound(
-        gctx, *refSurface, freePars.value(), covCache);
-    auto targetCov = std::get<Acts::BoundSquareMatrix>(varNewCov);
-
     Acts::ActsSquareMatrix<6> J = jacobianToEdm4hep(
         targetPars[eBoundTheta], targetPars[eBoundQOverP], Bz);
-    Acts::ActsSquareMatrix<6> cIn = targetCov.template topLeftCorner<6, 6>();
-    result.covariance = J * cIn * J.transpose();
+    result.covariance = J * params.covariance().value() * J.transpose();
```
Comment (andiwand) [inline Plugins/EDM4hep/src/EDM4hepUtil.cpp:150]: maybe I overlooked something but I think we can just directly use the bound covariance here @paulgessinger ?

File: Plugins/EDM4hep/src/EDM4hepUtil.cpp [hunk 2]
```diff
@@ -133,36 +129,31 @@ Parameters convertTrackParametersToEdm4hep(const Acts::GeometryContext& gctx,
...
-    auto [varNewCov, varNewJac] = Acts::transportCovarianceToBound(
-        gctx, *refSurface, freePars.value(), covCache);
-    auto targetCov = std::get<Acts::BoundSquareMatrix>(varNewCov);
-
+  if (targetCov) {
     Acts::ActsSquareMatrix<6> J = jacobianToEdm4hep(
         targetPars[eBoundTheta], targetPars[eBoundQOverP], Bz);
-    Acts::ActsSquareMatrix<6> cIn = targetCov.template topLeftCorner<6, 6>();
-    result.covariance = J * cIn * J.transpose();
+    result.covariance = J * targetCov.value() * J.transpose();
```
Comment (andiwand) [inline Plugins/EDM4hep/src/EDM4hepUtil.cpp:156]: @paulgessinger I tried to implement what we discussed. not sure if this is correct tho :smile:

**Review by paulgessinger — APPROVED** [review]
ok let's do it

**Comment by acts-project-service** [comment]
## :red_circle: Athena integration test results [7c9479d3c88b36bc89f19a8a8db12a946ed4ef78]
### Build job with this PR failed!
Please investigate the [build job](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34780012) for the [pipeline](https://gitlab.cern.ch/acts/acts-athena-ci/-/pipelines/6626716)!

**Pull request merged**